### PR TITLE
change parameters for unbound tuning

### PR DESCRIPTION
**Pull request by terassyi**
I changed `etc/unbound.yml` and `ignitions/common/files/etc/sysctl.d/70-cybozu.conf`.
This PR changes unbound configurations and sysctl parameters to tune unbound.

Signed-off-by: terasihma <tomoya-terashima@cybozu.co.jp>

### CODE REVIEW

File: ignitions/common/files/etc/sysctl.d/70-cybozu.conf [hunk 1]
```diff
@@ -12,6 +12,9 @@ net.ipv4.neigh.default.gc_thresh3 = 4096
 net.ipv4.tcp_keepalive_time = 600
 # Increase the maximum TCP send buffer size settable with setsockopt(2).
 net.core.wmem_max = 16777216
+# Increase the maximum TCP receive buffer size
+net.core.rmem_max = 4194304
+rmem_default = 4194304
```
Comment (ymmt2005) [inline ignitions/common/files/etc/sysctl.d/70-cybozu.conf:17]: Did you try this on your Ubuntu?
It seems that the correct name is `net.core.rmem_default`.
https://docs.kernel.org/admin-guide/sysctl/net.html?highlight=rmem_default#proc-sys-net-core-network-core-options

Comment (terassyi) [reply ignitions/common/files/etc/sysctl.d/70-cybozu.conf]: I'm sorry, but I forgot to check it.

File: etc/unbound.yml [hunk 2]
```diff
@@ -28,11 +28,21 @@ data:
...
+      num-threads: 4
+      incoming-num-tcp: 2240
+      outgoing-num-tcp: 1000
+      num-queries-per-thread: 4096
+      outgoing-range: 8192
       rrset-roundrobin: yes
       minimal-responses: yes
       extended-statistics: yes
       rrset-cache-size: 256m
+      rrset-cache-slabs: 2
```
Comment (ysksuzuki) [inline etc/unbound.yml:40]: Would you be able to see https://github.com/cybozu-go/cke/pull/567#discussion_r974869584 and modify these parameters accordingly?

**Review by ymmt2005 — COMMENTED** [review]
@terassyi @ysksuzuki 
I'm not pretty sure about these settings being applied for all services.
Aren't they specific to Unbound?

If so, I'd suggest customizing sysctl values only for Unbound pods.
https://kubernetes.io/docs/tasks/administer-cluster/sysctl-cluster/#setting-sysctls-for-a-pod

File: etc/unbound.yml (hunk 2 above)
Comment (terassyi) [reply etc/unbound.yml:40]: Should we change `incoming-num-tcp: 1000` according to its pull request?

https://github.com/cybozu-go/cke/blob/2f299cb7bf6bb689b71889a442fcaafa4261883b/op/nodedns/nodedns.go#L36

**Comment by ysksuzuki** [comment]
@ymmt2005 
Do you mean that it might be better to apply the changes of the parameters `net.core.rmem_max` and `net.core.rmem_default` only to unbound pods? If so, I think we probably should do it, because we change these parameters for unbound following the instruction https://unbound.docs.nlnetlabs.nl/en/latest/topics/performance.html

File: etc/unbound.yml (hunk 2 above)
Comment (ysksuzuki) [reply etc/unbound.yml:40]: I don't think so. Would you tell me why do think we need to change it?

**Comment by ymmt2005** [comment]
Correct. But chances are there that those settings might benefit other services in general.
Applying the settings only to Unbound pods is safe, anyway.

File: etc/unbound.yml (hunk 2 above)
Comment (terassyi) [reply etc/unbound.yml:40]: I also don't think that we need to change it.
I just wanted to confirm if we had to change.

**Comment by ysksuzuki** [comment]
@ymmt2005 We also run unbound containers in node-dns pods and those pods are `hostNetwork: true`. If we apply the changes of net.core.rmem_max and net.core.rmem_default for node-dns as well, I think we need to change them globally. What do you think?

**Comment by ymmt2005** [comment]
@ysksuzuki 
Unbound for node-local DNS serves only for a few pods and receives requests over UDP.
So, these settings won't make sense.

**Comment by ysksuzuki** [comment]
Right, those extend the TCP  receive buffer size. It doesn't make sense then.

**Comment by ymmt2005** [comment]
@ysksuzuki @terassyi 
I revisited these parameters and thought that having `net.core.rmem_max = 4194304` as a global setting is worth doing.
This setting does allow applications to have a 4MiB receive buffer at maximum, which is good for modern network apps.

Setting the default to 4 MiB would waste much memory, I think.

**Comment by ymmt2005** [comment]
@ysksuzuki correct!

**Review by ymmt2005 — APPROVED** [review]
LGTM.
I'd suggest telling other teams about the settings for larger receive buffer size might be worth considering if they are running a busy network server such as Nginx.

**Review by ysksuzuki — APPROVED** [review]

**Pull request merged**